tests/integration/fixtures.py

```diff
@@ -0,0 +1,196 @@
+
+    with ops_test.model_context(COS_MODEL_NAME):
+        await ops_test.model.deploy(  # type: ignore[union-attr]
+            entity_url="https://charmhub.io/cos-lite",
+            trust=True,
+        )
+        await ops_test.model.wait_for_idle(  # type: ignore[union-attr]
+            apps=[*ops_test.model.applications],  # type: ignore[union-attr]
+            raise_on_error=False,
+            status="active",
+            timeout=600,
+        )
+        await ops_test.model.create_offer(  # type: ignore[union-attr]
+            endpoint="prometheus:receive-remote-write",
+            offer_name="prometheus",
+            application_name="prometheus",
+        )
+
+
+async def deploy_cos_configuration(ops_test: OpsTest):
+    """Deploys `cos-configuration-k8s` charm.
+
+    Args:
+        ops_test: OpsTest
+    """
+    with ops_test.model_context(COS_MODEL_NAME):
+        await ops_test.model.deploy(  # type: ignore[union-attr]
+            "cos-configuration-k8s",
+            config={
+                "git_repo": "https://github.com/canonical/sdcore-cos-configuration",
+                "git_branch": "main",
+                "git_depth": 1,
+                "grafana_dashboards_path": "grafana_dashboards/sdcore/",
+            },
+        )
+        await ops_test.model.add_relation("cos-configuration-k8s", "grafana")  # type: ignore[union-attr]  # noqa: E501
+        await ops_test.model.wait_for_idle(  # type: ignore[union-attr]
+            apps=["cos-configuration-k8s"],
+            raise_on_error=False,
+            status="active",
+            timeout=300,
+        )
+
+
+def _create_subscriber(webui_ip: str) -> requests.Response:
+    """Creates a subscriber from a template.
+
+    Args:
+        webui_ip: IP address of the `webui` application unit
+
+    Returns:
+        requests.Response: requests.Response Object
+    """
+    template = JINJA_ENV.get_template("subscriber.json.j2")
+    subscriber_data = template.render(imsi=TEST_IMSI)
+    url = f"http://{webui_ip}:5000/api/subscriber/imsi-{TEST_IMSI}"
+    return requests.post(url=url, data=subscriber_data)
+
+
+def _create_device_group(webui_ip: str) -> requests.Response:
+    """Creates a device group from a template.
+
+    Args:
+        webui_ip: IP address of the `webui` application unit
+
+    Returns:
+        requests.Response: requests.Response Object
+    """
+    template = JINJA_ENV.get_template("device-group.json.j2")
+    device_group_json = json.loads(template.render(imsi=TEST_IMSI))
+    url = f"http://{webui_ip}:5000/config/v1/device-group/{TEST_DEVICE_GROUP_NAME}"
+    return requests.post(url, json=device_group_json)
+
+
+def _create_network_slice(webui_ip: str) -> requests.Response:
+    """Creates a network slice from a template.
+
+    Args:
+        webui_ip: IP address of the `webui` application unit
+
+    Returns:
+        requests.Response: requests.Response Object
+    """
+    template = JINJA_ENV.get_template("network-slice.json.j2")
```

Comment:
I thought the same about the templates :D It makes the python file shorter, because we keep non-python stuff separately.
I can change that to dicts if you want tho.